### ISSUE TITLE
danger-swiftlint doesn't warn, while SwiftLint itself does

### ISSUE THREAD

**Issue by freak4pc** [ashfurrow/danger-ruby-swiftlint#48]
I'm not sure if this is some misconfiguration (mainly due to the fact there's barely any configuration around danger-swiftlint... )

But when SwiftLint runs by itself, i see specifically
```
Todo Violation: TODOs should be avoided (...). (todo)
```

But when danger-swiftlint runs in my CI (Buddybuild), it doesn't warn for some odd reason
```
Installing danger 5.3.3
Fetching danger-swiftlint 0.7.0
Installing danger-swiftlint 0.7.0 with native extensions
Bundle complete! 2 Gemfile dependencies, 22 gems now installed.
Bundled gems are installed into /tmp/sandbox/workspace/vendor/bundle.

Running Danger
Using Command Line: bundle exec danger --dangerfile=Dangerfile

Results:
    Warnings:
    - [ ] Big PR
    - [ ] The `Podfile` was updated
```

But no specific warnings related to SwiftLint, and I'm not exactly sure why or if I'm missing something.

Gemfile only has danger and danger-swiftlint

The portion of `Dangerfile` related to SwiftLint is: 
```ruby
# SwiftLint
swiftlint.config_file = '.swiftlint.yml'
swiftlint.binary_path = './Pods/SwiftLint/swiftlint'
swiftlint.lint_files
swiftlint.lint_files inline_mode: true
```

I'm wondering if this is related to the fact I'm using a pinned version with CocoaPods, but I don't see why that would be an issue 🤔

Appreciate any help on this :)

**Comment by freak4pc** [comment]
Pushing another PR right after that one made it work, sorry about the confusion. 

I am having a different issue of danger-swiftlint ignoring my `excludes` statement in `.swiftlint.yml` which goes through all of my Pods and spits hundreds of warnings but that's probably a separate issue.

**Comment by ashfurrow** [comment]
Hmm, that's an odd one. Sorry you're experiencing this problem. I agree CocoaPods doesn't seem like a likely source of the problem. I'm not sure what's going on here – I'll try to take a look this weekend. If you could reproduce the problem in a test project, that would make debugging the issue a lot easier. Let me know if you make any progress!

**Comment by freak4pc** [comment]
Thanks for the reply @ashfurrow :) I'll try digging into it this weekend as well and provide some more feedback

**Comment by RishabhTayal** [comment]
I am having this issue as well. In my case I am using multiple `swiftlint.lint_files` for different project within the workspace. However I don't think that's what causing the issue.

**Comment by ashfurrow** [comment]
Could this be related to #49?

**Comment by RishabhTayal** [comment]
I am not getting any error when running swiflint but seems like when running `swiftlint` command for multiple configuration files doesn't comment on the pr for remaining projects.

**Comment by freak4pc** [comment]
Still couldn't dive much into this as i'm swamped with work but noticed much of this in my PR logs which is interesting 🤔 

It seems as if it's using the swift filenames as folders for some odd reason ... 

Also wondering on why it's using @marcelofabri 's user folder... ❓ 

```

Using Command Line: bundle exec danger --dangerfile=Dangerfile

fatal error: Could not read configuration file at path '/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml': file /Users/marcelofabri/Westfield/SwiftLint/Source/SwiftLintFramework/Models/Configuration.swift, line 167
/tmp/sandbox/workspace/TimHortons/RBIKit/API/THPayments.swift/.swiftlint.yml:File not found.

fatal error: Could not read configuration file at path '/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml': file /Users/marcelofabri/Westfield/SwiftLint/Source/SwiftLintFramework/Models/Configuration.swift, line 167
/tmp/sandbox/workspace/TimHortons/RBIKit/Common/RBIUserProfile.swift/.swiftlint.yml:File not found.

fatal error: Could not read configuration file at path '/tmp/sandbox/workspace/TimHortons/RBIKit/Models/THGiftCard.swift/.swiftlint.yml': file /Users/marcelofabri/Westfield/SwiftLint/Source/SwiftLintFramework/Models/Configuration.swift, line 167
/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml:File not found.

fatal error: Could not read configuration file at path '/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml': file /Users/marcelofabri/Westfield/SwiftLint/Source/SwiftLintFramework/Models/Configuration.swift, line 167
/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml:File not found.

/tmp/sandbox/workspace/TimHortons/SomeSwiftFile.swift/.swiftlint.yml:File not found.

**Comment by marcelofabri** [comment]
That's just a side effect on how `fatalError` works in Swift. It uses `#file`, which is resolved in compilation time. And you see that path because the CocoaPods build was compiled on my machine.

Note that it's only the path for the Swift file where the `fatalError` was called. The important part is the message after `fatal error:`, before `file:`. 

This can be related to #49. I recommend you to update SwiftLint to 0.21.0.

**Comment by Sega-Zero** [comment]
Any news on this? swiftlint 0.22.0, still not working

**Comment by ashfurrow** [comment]
Hey, sorry no updates. It would be really helpful if anyone were able to put together a sample project demonstrating the problem, so we can reproduce and debug.

**Comment by Sega-Zero** [comment]
Will try to create a sample repo asap

**Comment by Sega-Zero** [comment]
Take a look at https://github.com/Sega-Zero/danger-sample/pull/1
It should fail, but it shows all green.
Right now I have to use this Dangerfile to make things working, but that's not that comfortable

```ruby
warn("PR is classed as Work in Progress") if github.pr_title.include? "[WIP]"

message("Started linting a PR " + github.pr_title)
system("./project/Pods/SwiftLint/swiftlint --strict --quiet --config .swiftlint.yml --path ./project/" )
fail("Linting failed") if $?.exitstatus != 0

message("I'm done with this PR")
```

**Comment by ashfurrow** [comment]
Okay, thanks! I've used your project to reproduce the bug – I think it's related to the config file paths not being expanded properly. I've fixed the bug and introduced a new verbose logging mode in #60, which has been released as `danger-swift` 0.10.1. If you could, please update to this version and enable verbose logging so we can get a better idea of what's going on.

**Comment by heidiproske** [comment]
hey @ashfurrow I was taking a look at @Sega-Zero's code snippet from Oct 7 while I was trying to debug something with `danger local`. If I use his snippet it'll lint *all* files vs when I run danger-swiftlint it only lints modified/added files. 

Is there a way to lint all files with danger-swiftlint? And not just those modified in a PR?

**Comment by heidiproske** [comment]
hey @ashfurrow I was taking a look at @Sega-Zero's code snippet from Oct 7 while I was trying to debug something with `danger local`. If I use his snippet it lints *all* files (and I get the results I want).

But when I run danger-swiftlint it only lints modified/added files for the PR. 

My Dangerfile looks like this:
```
# Run SwiftLint
swiftlint.config_file = './.swiftlint.yml'
swiftlint.verbose = true
swiftlint.lint_files fail_on_error: false
swiftlint.lint_files inline_mode: true
```

Is there a way for danger-swiftlint to show all outstanding warnings in my project on every PR - not just the files I touched? (I did try adding the `included:` section to my `.swiftlint.yml`). I can't see anything in the docs.

**Comment by Sega-Zero** [comment]
@heidiproske, @ashfurrow [answered](https://github.com/Sega-Zero/danger-sample/pull/1#discussion_r143359175) the same question from me.

**Comment by omirho** [comment]
This seems to have been fixed with #60 
Closing.

**Comment by Sega-Zero** [comment]
This is not really fixed, but explained in the link I mentioned above. It is by design.
Those who came here for a same question, I recommend to switch to [danger-swiftlint](https://github.com/ashfurrow/danger-swiftlint) written in swift. It is great!
